crates/sema/src/ast_lowering/resolve.rs

```diff
@@ -196,6 +196,84 @@ impl super::LoweringContext<'_, '_, '_> {
         }
     }
 
+    #[instrument(level = "debug", skip_all)]
+    pub(super) fn check_base_ctor_args(&self) {
+        for main_contract in self.hir.contracts() {
```

Comment:
why is this using the AST instead of being a HIR check like the other ones you did? this shouldnt happen during lowering